flask_resty/filtering.py

```diff
@@ -52,79 +79,119 @@ def format_validation_error(self, message):
     def get_field(self, view):
         raise NotImplementedError()
 
-    def get_filter_clause(self, view, value):
+    def get_filter_arg(self, view):
         raise NotImplementedError()
 
 
-class ColumnFilterField(FilterFieldBase):
-    def __init__(self, column_name, operator, **kwargs):
-        super(ColumnFilterField, self).__init__(**kwargs)
+class ColumnFilter(FieldFilter):
+    def __init__(
+        self,
+        column_name=None,
+        operator=None,
+        required=False,
+        **kwargs
+    ):
+        if operator is None and callable(column_name):
+            operator = column_name
+            column_name = None
+
+        if not operator:
+            raise TypeError("must specify operator")
+
+        super(ColumnFilter, self).__init__(operator, **kwargs)
+
+        self._has_explicit_column_name = column_name is not None
         self._column_name = column_name
-        self._operator = operator
+        self._required = required
+
+    def maybe_set_arg_name(self, arg_name):
+        if self._has_explicit_column_name:
+            return
+
+        if self._column_name and self._column_name != arg_name:
+            raise TypeError(
+                "cannot use ColumnFilter without explicit column name " +
+                "for multiple arg names",
+            )
+
+        self._column_name = arg_name
+
+    def filter_query(self, query, view, arg_value):
+        # Missing value handling on the schema field is not relevant here.
+        if arg_value is None:
+            if not self._required:
+                return query
+
+            raise ApiError(400, {'code': 'invalid_filter.missing'})
+
+        return super(ColumnFilter, self).filter_query(query, view, arg_value)
 
     def get_field(self, view):
         return view.deserializer.fields[self._column_name]
 
-    def get_filter_clause(self, view, value):
-        column = getattr(view.model, self._column_name)
-        return self._operator(column, value)
+    def get_filter_arg(self, view):
+        return getattr(view.model, self._column_name)
 
 
-class ModelFilterField(FilterFieldBase):
+class ModelFilter(FieldFilter):
     def __init__(self, field, filter, **kwargs):
-        super(ModelFilterField, self).__init__(**kwargs)
+        super(ModelFilter, self).__init__(filter, **kwargs)
+
         self._field = field
-        self._filter = filter
 
     def get_field(self, view):
         return self._field
 
-    def get_filter_clause(self, view, value):
-        return self._filter(view.model, value)
+    def get_filter_arg(self, view):
+        return view.model
+
+
+# -----------------------------------------------------------------------------
+
+
+def model_filter(field, **kwargs):
```

Comment:
This used to be `@filter_function`. I'm 60/40 on renaming this. This is the only part of the API that's changing that we actually use. Nothing else here imposes changes on most application code.